src/main/java/com/ivanmostovyi/demo/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,52 @@
+package com.ivanmostovyi.demo.config;
+
+import com.ivanmostovyi.demo.domain.UserRole;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    private UserDetailsService userDetailsService;
+
+    private PasswordEncoder passwordEncoder;
+
+    public WebSecurityConfig(@Qualifier("userServiceImpl") UserDetailsService userDetailsService,
+                             PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                    .antMatchers("/user/new").permitAll()
+                    .anyRequest().hasAuthority(UserRole.ROLE_USER.name())
+                    .and()
+                .formLogin()
+                    .loginPage("/login")
+                    .defaultSuccessUrl("/message/inbox")
+                    .failureForwardUrl("/login?error=true")
```

Comment:
`failureForwardUrl` should not be overridden here.
Default behaviour will be that one, that you need